go/multipooler/manager/rpc_backup.go

```diff
@@ -300,6 +300,21 @@ func (pm *MultiPoolerManager) getBackupsLocked(ctx context.Context, limit uint32
 		return nil, err
 	}
 
+	backups, err := pm.listBackups(ctx)
+	if err != nil {
+		return nil, err
+	}
+
+	// Apply limit if specified
+	if limit > 0 && uint32(len(backups)) > limit {
+		backups = backups[:limit]
+	}
+
+	return backups, nil
+}
+
+// listBackups retrieves backup metadata from pgbackrest.
+func (pm *MultiPoolerManager) listBackups(ctx context.Context) ([]*multipoolermanagerdata.BackupMetadata, error) {
```

Comment:
It seems this is called in context where action lock i held, but not obvious is pm.mu  is also locked. 

Should we make sure that lock is held?

Reply:
I don't *think* pm.mu needs to be locked, and I've added a comment explaining why. let me know what you think